### PR TITLE
RzCore: add scr.prompt.flag.only config var

### LINKED ISSUE #866: Show current flag in the command prompt, near the address
**Is your feature request related to a problem? Please describe.**
For example, I set `db sym.imp.free` and `db sym.imp.malloc` (i have no idea where they are in memory). Then I do `dc` to continue debugging and I hit a breakpoint. I see this:
```
[0x55b934e5ad80]> dc
hit breakpoint at: 0x55b934e5aae0
[0x55b934e5aae0]> 
```
I have no idea what I hit.

**Describe the solution you'd like**
It would be great to see something like this:
```
[0x55b934e5ad80:main]> dc
hit breakpoint at: 0x55b934e5aae0
[0x55b934e5aae0:sym.imp.malloc]> 
```

**Describe alternatives you've considered**
```
[0x55b934e5ad80:main]> dc
hit breakpoint at: 0x55b934e5aae0
[0x55b934e5aae0:malloc]> 
```

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Example:
```
[0x000006a0]> e scr.prompt.file=true
[hello_world:0x000006a0]> e scr.prompt.flag=true
[hello_world:entry0:0x000006a0]> sd 1
[hello_world:entry0 + 1:0x000006a0]> e scr.prompt.flag.only=true
[hello_world:entry0 + 1]> e scr.prompt.sect=true
[hello_world:.text:entry0 + 1]> 
```

**Test plan**

...

**Closing issues**

Fixes #866

### CODE REVIEW

**Comment by XVilka** [comment]
I am not sure `hello_world:` part is needed. What about hiding it?

**Comment by ret2libc** [comment]
> I am not sure `hello_world:` part is needed. What about hiding it?

It is just to show a full example. hello_world is the name of the binary and it is shown only when scr.prompt.file=true.

Before it was like this:
```
[0x00005e50]> e scr.prompt.file=true
"ls"[0x00005e50]> 
```

And I find it not that good to look at.

**Review by XVilka — CHANGES_REQUESTED** [review]
Please add tests. The idea looks good.

**Comment by ret2libc** [comment]
> Please add tests. The idea looks good.

I'm having troubles capturing the prompt in the tests actually...

File: librz/core/core.c [hunk 1]
```diff
@@ -2731,56 +2731,70 @@ RZ_API void rz_core_prompt_loop(RzCore *r) {
...
-static void chop_prompt(const char *filename, char *tmp, size_t max_tmp_size) {
-	size_t tmp_len, file_len;
-	unsigned int OTHRSCH = 3;
-	const char DOTS[] = "...";
-	int w, p_len;
+static bool prompt_add_offset(RzCore *core, RzStrBuf *sb, bool add_sep) {
+	if (add_sep) {
+		rz_strbuf_append(sb, ":");
+	}
+	if (rz_config_get_i(core->config, "scr.prompt.flag")) {
```
Comment (XVilka) [inline librz/core/core.c]: Use `rz_config_get_b()` for booleans.

File: librz/core/core.c [hunk 2]
```diff
@@ -2731,56 +2731,70 @@ RZ_API void rz_core_prompt_loop(RzCore *r) {
...
+		rz_strbuf_append(sb, ":");
 	}
+	rz_strbuf_append(sb, rz_file_basename(core->io->desc->name));
 	return true;
 }
 
-static void prompt_sec(RzCore *r, char *s, size_t maxlen) {
-	const RzBinSection *sec = rz_bin_get_section_at(rz_bin_cur_object(r->bin), r->offset, true);
+static bool prompt_add_section(RzCore *core, RzStrBuf *sb, bool add_sep) {
+	if (!rz_config_get_i(core->config, "scr.prompt.sect")) {
```
Comment (XVilka) [inline librz/core/core.c]: `rz_config_get_b()`?

File: librz/core/core.c [hunk 3]
```diff
@@ -2731,56 +2731,70 @@ RZ_API void rz_core_prompt_loop(RzCore *r) {
...
-		size_t dots_size = sizeof(DOTS);
-		size_t chop_point = (size_t)(p_len - OTHRSCH - file_len - dots_size - 1);
-		if (chop_point < (max_tmp_size - dots_size - 1)) {
-			tmp[chop_point] = '\0';
-			strncat(tmp, DOTS, dots_size);
+			rz_strbuf_append(sb, ":");
 		}
 	}
+
+	if (rz_config_get_i(core->config, "asm.segoff")) {
```
Comment (XVilka) [inline librz/core/core.c]: It's probably also a boolean, right?

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by kazarmy — COMMENTED** [review]
> I'm having troubles capturing the prompt in the tests actually...

Technically I think you can:
https://github.com/rizinorg/rizin/blob/a1c2ed18d762d3641e6690e0cb6b58b45f1c3609/test/db/tools/rz#L183-L188
but due to the way the prompt works, there will probably be a ton of test output.

**Comment by kazarmy** [comment]
Or you can just set config vars using `-e` and see what the initial prompt is.

**Comment by ret2libc** [comment]
Thanks @kazarmy ! @XVilka I added the test and fixed the comments, please review again.

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**